### PR TITLE
Use resized images for post thumbnails

### PR DESCRIPTION
**Pull request by benoitchantre**
## Description

Addresses issues #468 and #453.

## List of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

File: inc/Post_Thumbnails/Component.php [hunk 1]
```diff
@@ -40,4 +42,12 @@ public function initialize() {
 	public function action_add_post_thumbnail_support() {
 		add_theme_support( 'post-thumbnails' );
 	}
+
+	/**
+	 * Adds custom image sizes.
+	 */
+	public function action_add_image_sizes() {
+		set_post_thumbnail_size( 720, 308, true );
```
Comment (mor10) [inline inc/Post_Thumbnails/Component.php]: Use `add_image_size()` instead. Many plugins expect there to be a standard 150 x 150 thumbnail available. `set_post_thumbnail_size()` changes this default. Best-practice is to always preserve the standard core sizes and include custom sizes where other sizes are required.

Comment (benoitchantre) [reply inc/Post_Thumbnails/Component.php]: Thank you for the review. I have updated the code and squashed the commits.

File: inc/Post_Thumbnails/Component.php [hunk 2]
```diff
@@ -40,4 +42,12 @@ public function initialize() {
 	public function action_add_post_thumbnail_support() {
 		add_theme_support( 'post-thumbnails' );
 	}
+
+	/**
+	 * Adds custom image sizes.
+	 */
+	public function action_add_image_sizes() {
+		add_image_size( 'wp-rig-post-thumbnail', 720, 308, true );
```
Comment (bamadesigner) [inline inc/Post_Thumbnails/Component.php]: Let's use the built in post thumbnail function, which we can set the size with `set_post_thumbnail_size()`.

And then use `'post-thumbnail'` as the size.

File: template-parts/content/entry_thumbnail.php [hunk 3]
```diff
@@ -46,7 +46,7 @@
 			);
 		} else {
 			the_post_thumbnail(
-				'post-thumbnail',
+				'wp-rig-post-thumbnail',
```
Comment (bamadesigner) [inline template-parts/content/entry_thumbnail.php]: Let's keep this instance as "post-thumbnail".

File: template-parts/content/entry_thumbnail.php [hunk 4]
```diff
@@ -34,7 +34,7 @@
 		global $wp_query;
 		if ( 0 === $wp_query->current_post ) {
 			the_post_thumbnail(
-				'full',
+				'wp-rig-post-thumbnail-tall',
```
Comment (bamadesigner) [inline template-parts/content/entry_thumbnail.php]: Use "wp-rig-featured".

File: template-parts/content/entry_thumbnail.php [hunk 5]
```diff
@@ -24,7 +24,7 @@
 if ( is_singular( get_post_type() ) ) {
 	?>
 	<div class="post-thumbnail">
-		<?php the_post_thumbnail( 'full', array( 'class' => 'skip-lazy' ) ); ?>
+		<?php the_post_thumbnail( 'wp-rig-post-thumbnail', array( 'class' => 'skip-lazy' ) ); ?>
```
Comment (bamadesigner) [inline template-parts/content/entry_thumbnail.php]: Use "wp-rig-featured".

File: inc/Post_Thumbnails/Component.php (hunk 2 above)
Comment (bamadesigner) [reply inc/Post_Thumbnails/Component.php]: So, after discussion, we only need one new image size: ""wp-rig-featured".

**Comment by benoitchantre** [comment]
@bamadesigner Should be ok now.

**Comment by mor10** [comment]
@bamadesigner Can you clear your objection and pass this through?

File: template-parts/content/entry_thumbnail.php [hunk 6]
```diff
@@ -24,7 +24,7 @@
 if ( is_singular( get_post_type() ) ) {
 	?>
 	<div class="post-thumbnail">
-		<?php the_post_thumbnail( 'full', [ 'class' => 'skip-lazy' ] ); ?>
+		<?php the_post_thumbnail( 'wp-rig-post-thumbnail', [ 'class' => 'skip-lazy' ] ); ?>
```
Comment (bamadesigner) [inline template-parts/content/entry_thumbnail.php]: Needs to be "wp-rig-featured", not "wp-rig-post-thumbnail".

File: template-parts/content/entry_thumbnail.php [hunk 7]
```diff
@@ -46,7 +46,7 @@
 			);
 		} else {
 			the_post_thumbnail(
-				'post-thumbnail',
+				'wp-rig-featured',
```
Comment (bamadesigner) [inline template-parts/content/entry_thumbnail.php]: Needs to be "post-thumbnail", not "wp-rig-featured".

**Comment by bamadesigner** [comment]
@mor10 Thoughts on image size names?

**Comment by benoitchantre** [comment]
@mor10 Should the posts use the same image size or have a specific image size for the first post? I remember this has been discussed but can't find anything written.

File: template-parts/content/entry_thumbnail.php (hunk 6 above)
Comment (benoitchantre) [reply template-parts/content/entry_thumbnail.php]: Fixed. Thank you for having noticed this mistake.

**Comment by mor10** [comment]
For singular posts, the featured image should be quite large. For indexes, the images should be confined.

File: inc/Post_Thumbnails/Component.php [hunk 8]
```diff
@@ -40,4 +42,12 @@ public function initialize() {
 	public function action_add_post_thumbnail_support() {
 		add_theme_support( 'post-thumbnails' );
 	}
+
+	/**
+	 * Adds custom image sizes.
+	 */
+	public function action_add_image_sizes() {
+		add_image_size( 'wp-rig-featured', 720, 480, true );
+		add_image_size( 'wp-rig-thumbnail', 720, 308, true );
```
Comment (bamadesigner) [inline inc/Post_Thumbnails/Component.php]: @benoitchantre We don't need our own thumbnail size. Where `wp-rig-thumbnail` is used, we should be using just `post-thumbnail`.

Comment (bamadesigner) [reply inc/Post_Thumbnails/Component.php]: @benoitchantre I will take care of this so we can merge for the release.

**Pull request merged**